boards/grand_central_m4/examples/usb_serial.rs

```diff
@@ -14,67 +14,63 @@
 //! Note leds may appear white during debug. Either build for release or add
 //! opt-level = 2 to profile.dev in Cargo.toml
 
-use bsp::hal;
 use grand_central_m4 as bsp;
 
+use bsp::hal;
+
 #[cfg(not(feature = "use_semihosting"))]
 use panic_halt as _;
 #[cfg(feature = "use_semihosting")]
 use panic_semihosting as _;
 
-use bsp::entry;
-use cortex_m::interrupt::free as disable_interrupts;
+use cortex_m::asm::delay as cycle_delay;
 use cortex_m::peripheral::NVIC;
-use hal::clock::GenericClockController;
-use hal::pac::{interrupt, CorePeripherals, Peripherals};
-use hal::prelude::*;
-use hal::time::Hertz;
-use hal::timer::TimerCounter;
-use hal::timer_traits::InterruptDrivenTimer;
-use hal::usb::UsbBus;
-use smart_leds::{colors, hsv::RGB8, SmartLedsWrite};
 use usb_device::bus::UsbBusAllocator;
 use usb_device::prelude::*;
 use usbd_serial::{SerialPort, USB_CLASS_CDC};
-use ws2812_timer_delay as ws2812;
+
+use bsp::clock::gclk::Gclk1Id;
+use bsp::entry;
+use bsp::pin_alias;
+use hal::clock::v2::clock_system_at_reset;
+use hal::pac::{interrupt, CorePeripherals, Peripherals};
+use hal::prelude::*;
+use hal::usb::UsbBus;
 
 #[entry]
 fn main() -> ! {
     let mut peripherals = Peripherals::take().unwrap();
     let mut core = CorePeripherals::take().unwrap();
-    let mut clocks = GenericClockController::with_internal_32kosc(
+    let (mut buses, clocks, tokens) = clock_system_at_reset(
+        peripherals.oscctrl,
+        peripherals.osc32kctrl,
         peripherals.gclk,
-        &mut peripherals.mclk,
-        &mut peripherals.osc32kctrl,
-        &mut peripherals.oscctrl,
+        peripherals.mclk,
         &mut peripherals.nvmctrl,
     );
-    let gclk0 = clocks.gclk0();
-    let tc2_3 = clocks.tc2_tc3(&gclk0).unwrap();
-    let mut timer = TimerCounter::tc3_(&tc2_3, peripherals.tc3, &mut peripherals.mclk);
-    InterruptDrivenTimer::start(&mut timer, Hertz::MHz(3).into_duration());
 
     let pins = bsp::Pins::new(peripherals.port);
-    let neopixel_pin = pins.neopixel.into_push_pull_output();
-    let mut neopixel = ws2812::Ws2812::new(timer, neopixel_pin);
-    let _ = neopixel.write((0..5).map(|_| RGB8::default()));
+    let mut red_led: bsp::RedLed = pin_alias!(pins.red_led).into();
 
     let bus_allocator = unsafe {
         USB_ALLOCATOR = Some(bsp::usb_allocator(
             peripherals.usb,
-            &mut clocks,
-            &mut peripherals.mclk,
+            clocks.dfll,
+            tokens.gclks.gclk1,
+            tokens.pclks.usb,
+            clocks.ahbs.usb,
+            buses.apb.enable(tokens.apbs.usb),
             pins.usb_dm,
             pins.usb_dp,
         ));
         USB_ALLOCATOR.as_ref().unwrap()
     };
 
     unsafe {
-        USB_SERIAL = Some(SerialPort::new(&bus_allocator));
+        USB_SERIAL = Some(SerialPort::new(bus_allocator));
```

Comment:
This appears to be a safe function. If so, please move it outside of the unsafe block.